flink/src/main/java/org/apache/iceberg/flink/source/FlinkSplitPlanner.java

```diff
@@ -22,33 +22,58 @@
 import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.util.List;
+import org.apache.flink.annotation.Internal;
 import org.apache.iceberg.CombinedScanTask;
 import org.apache.iceberg.Table;
 import org.apache.iceberg.TableProperties;
 import org.apache.iceberg.TableScan;
 import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.flink.source.split.IcebergSourceSplit;
 import org.apache.iceberg.io.CloseableIterable;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 
-class FlinkSplitGenerator {
-  private FlinkSplitGenerator() {
+@Internal
+public class FlinkSplitPlanner {
+  private FlinkSplitPlanner() {
   }
 
-  static FlinkInputSplit[] createInputSplits(Table table, ScanContext context) {
-    List<CombinedScanTask> tasks = tasks(table, context);
-    FlinkInputSplit[] splits = new FlinkInputSplit[tasks.size()];
-    for (int i = 0; i < tasks.size(); i++) {
-      splits[i] = new FlinkInputSplit(i, tasks.get(i));
+  static FlinkInputSplit[] planInputSplits(Table table, ScanContext context) {
```

Comment:
Why change the name of this method?

Reply:
`create/generate` implies creating sth new. This is actually plan/discover splits from table. Hence changed the method name. I actually also renamed the class name from `FlinkSplitGenerator` to `FlinkSplitPlanner`. This is an internal class. So it shouldn't break user code.

Reply:
`create/generate` implies creating sth new. This is actually plan/discover splits from table. Hence changed the method name. I actually also renamed the class name from `FlinkSplitGenerator` to `FlinkSplitPlanner`. This is an internal class. So it shouldn't break user code.